src/core/echarts.ts

```diff
@@ -418,11 +418,11 @@ class ECharts extends Eventful<ECEventDefinition> {
                 env.hasGlobalWindow ? window : global
             ) as any;
 
-            defaultRenderer = root.__ECHARTS__DEFAULT__RENDERER__ || defaultRenderer;
+            defaultRenderer = root?.__ECHARTS__DEFAULT__RENDERER__ ?? defaultRenderer;
```

Comment:
As the build target of ECharts is currently still `ES5`, to reduce the transpiled code size as possible,  instead of `?.` and `??`, just fallback `root` to `{}` when it is not defined or simply surround the block (line 421- line 428) with `if(root) {}`.

Reply:
You're right. I overlooked that. I'll revise it.